### PR TITLE
Add per endpoint cert

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Purpose

We receive endpoint_certificate.yaml in the endpoint-certifates directory of the api project. it has the following format. It will be used to pick the specific certificate for the endpoint.
 if the secure endpoint has not mentioned in the following file, other certificates in the endpoint-cerficate directory would be picked out. if that also not  provided, the endpoint socket configuration will use default truststore
```
type: endpoint_certificates
version: v4.0.0
data:
 -
  alias: myAlias
  endpoint: http://efg:9098/hello
  certificate: efg.crt

```

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2295

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Tested locally

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by pubudu538 — APPROVED** [review]

File: adapter/internal/discovery/xds/server.go [hunk 1]
```diff
@@ -362,8 +362,27 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, environments []string) (
...
 	updateVhostInternalMaps(apiYaml.ID, apiYaml.Name, apiYaml.Version, vHost, newLabels)
 
-	routes, clusters, endpoints := oasParser.GetRoutesClustersEndpoints(mgwSwagger, apiProject.UpstreamCerts,
-		apiProject.InterceptorCerts, vHost, organizationID)
+	// create cert map for API
+	certMap := make(map[string][]byte)
+	interceptCertMap := make(map[string][]byte)
+	if len(apiProject.EndpointCerts) > 0 && len(apiProject.UpstreamCerts) > 0 {
+		for url, certFile := range apiProject.EndpointCerts {
+			if certBytes, found := apiProject.UpstreamCerts[certFile]; found {
```
Comment (renuka-fernando) [inline adapter/internal/discovery/xds/server.go:370]: what if we log anerror, if cert not found

Comment (AmaliMatharaarachchi) [reply adapter/internal/discovery/xds/server.go:370]: fixed with https://github.com/wso2/product-microgateway/pull/2402/commits/5d1eb7bb4588efe49440493e74d1d46878d3fad4

**Review by renuka-fernando — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20211123.7

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211123.11

**Comment by choreo-cicd** [comment]
[succeeded] stage-deployment : 20211124.2

**Comment by choreo-cicd** [comment]
[succeeded] prod-deployment : 20211124.2

**Pull request merged**